openkfw/TruBudget
# Node Connection Status

### Checklist



- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/openkfw/TruBudget/blob/master/CONTRIBUTING.md).

### Description
 * Display connection status of each node in the network list (/nodes page)
 * Used `getpeerinfo` to retrieve all connected peers, compare with node list to determine status
 
### Test Plan
 1. in scripts/development/docker-compose.yml add following service right after *master-node*:
 ```
  slave-node:
    build:
      context: ../../blockchain
    environment:
      ORGANIZATION: ${ORGANIZATION}
      RPC_PORT: ${RPC_PORT}
      RPC_PASSWORD: ${RPC_PASSWORD}
      P2P_HOST: master-node
      P2P_PORT: ${P2P_PORT}
      API_PROTO: http
      API_HOST: master-api
      API_PORT: ${API_PORT}
      EXTERNAL_IP: master-node
    volumes:
      - /tmp/bcSlaveVolume:/root/.multichain
    ports:
      - "7448:7447"
 ```
 2. Run app as normal, approve new node and check that Network table on /nodes page **displays** two nodes with **Connection** status 
 3. In new terminal window run `docker exec -it trubudget-dev_slave-node_1 bash`
 4. In afterwards-opened subshell run `multichain-cli TrubudgetChain stop`
 5. Check frontend, one node should be disconnected
 6. Return to the subshell and run `multichaind -txindex -port=7447 -autosubscribe=streams TrubudgetChain@master-node:7447 “” -externalip=master-node`
 7. Refresh the page on the frontend. The node should be connected again




### Resources
 * [Multichain-CLI tutorial](https://github.com/openkfw/TruBudget/blob/master/doc/tutorials/multichain-cli/multichain-cli.md)

### Issue
Subtask of #746
